packages/sdk-react/src/components/approvalRequests/ApprovalRequestsTable/AutocompleteCreatedBy/AutocompleteCreatedBy.tsx

```diff
@@ -0,0 +1,98 @@
+import { useState } from 'react';
+
+import { components } from '@/api';
+import { useMoniteContext } from '@/core/context/MoniteContext';
+import { MoniteScopedProviders } from '@/core/context/MoniteScopedProviders';
+import { useRootElements } from '@/core/context/RootElementsProvider';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+import { Autocomplete, CircularProgress, TextField } from '@mui/material';
+
+interface AutocompleteCreatedByProps {
+  onChange: (id?: string) => void;
+}
+
+type EntityUserType = components['schemas']['EntityUserResponse'];
+
+export const AutocompleteCreatedBy = (props: AutocompleteCreatedByProps) => (
+  <MoniteScopedProviders>
+    <AutocompleteCreatedByBase {...props} />
+  </MoniteScopedProviders>
+);
+
+export const AutocompleteCreatedByBase = ({
+  onChange: onChangeFilter,
+}: AutocompleteCreatedByProps) => {
+  const { i18n } = useLingui();
+  const { api } = useMoniteContext();
+  const { root } = useRootElements();
+
+  const [open, setOpen] = useState<boolean>(false);
+  const [value, setValue] = useState<EntityUserType | null>(null);
+  const [inputValue, setInputValue] = useState('');
+
+  const {
+    data: users,
+    isLoading: isUsersLoading,
+    refetch,
+  } = api.entityUsers.getEntityUsers.useQuery(
+    {
+      query: { first_name: inputValue || undefined },
```

Comment:
I've noticed that the filtering doesn't work in a user-friendly enough way. For example, if you don't specify the name completely, the result will be empty:

<img width="278" alt="image" src="https://github.com/team-monite/monite-sdk/assets/725645/8281d06f-447d-4fbc-a3e1-6da71dd8d10f">

At the same time, full matching works correctly.

<img width="290" alt="image" src="https://github.com/team-monite/monite-sdk/assets/725645/ec18e799-0bc3-4196-a259-2b5b8dd32e50">

It looks like this API needs improvement because searching by full name is quite important. Could you create a task for the backend team?

Reply:
I sent a request to the platform team and asked to create a task